CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,32 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate Tax when Financial Type is changed.
+   *
+   * @param array $params
+   * @param float $totalAmount
+   * @param array $oldTaxAmounts
+   * @param float $changeFTAmount
+   *
+   */
+  public static function calculateTaxForChangeInFinancialType(
```

Comment:
Again, the same way as https://github.com/civicrm/civicrm-core/pull/9685, we have a confusion between what the method does and when it gets called. It looks like this method can be called just as fine even when the Financial Type have not changed